### PR TITLE
[Debt] Removes `hasDiploma`, `has_diploma`

### LINKED ISSUE #8661: ♻️ Remove deprecated `hasDiploma` field
## ♻️ Debt/Refactor

The `hasDiploma` field has been marked as deprecated for sometime and should be removed. 

## 🕵️ Details

The field exists and is referenced in various places, some places where it explicitly is set to do nothing like in filtering.
It ought to be removed

## 🙋‍♀️ Proposed Solution

Remove it
- from schema
- from PHP unit tests
- from snapshot generation
- from any front end instances, like the filters where it is disconnected from API
- Database
- Use caution

## ✅ Acceptance Criteria

- [ ] Removed from where decided in Proposed solution

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #8661.

## 👋 Introduction

This PR removes the `hasDiploma` field as well as any strings or components that are otherwise useless without it.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Run `refresh_api.sh`
2. Run fresh seed for database
3. `npm run build`
4. Ensure no references to `hasDiploma`, `has_diploma` (other than old migrations and tables from step 5) in the codebase
5. Ensure only `has_diploma` columns are for `applicant_filters` and `pool_candidate_filters` tables by running SQL query:
```
select table_name from INFORMATION_SCHEMA.COLUMNS 
where COLUMN_NAME like '%has_diploma%' 
order by TABLE_NAME
```

### CODE REVIEW

**Review by esizer — COMMENTED** [review]
Looking great! I just had one question about data retention.

File: api/database/migrations/2023_12_13_180500_drop_has_diploma_column.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('applicant_filters', function (Blueprint $table) {
+            $table->dropColumn('has_diploma');
+        });
+        Schema::table('pool_candidate_filters', function (Blueprint $table) {
+            $table->dropColumn('has_diploma');
+        });
```
[the remark this answers is not in the document]
Comment (mnigh) [reply api/database/migrations/2023_12_13_180500_drop_has_diploma_column.php:19]: i'm not sure, i was following the acceptance criteria that includes:

> Removed from where decided in Proposed solution
> 🙋‍♀️ Proposed Solution
> Remove it
> - Database

@tristan-orourke can you weigh in on whether or not these columns should be kept or dropped from the database?

[the remark this answers is not in the document]
Comment (petertgiles) [reply api/database/migrations/2023_12_13_180500_drop_has_diploma_column.php:19]: Decision from scrum: drop from users table, keep in requests tables

**Review by esizer — DISMISSED** [review]
Nice 🪓 ! Assuming the tests all pass, this LGTM! 👍

File: api/database/migrations/2023_12_13_180500_drop_has_diploma_column.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (mnigh) [reply api/database/migrations/2023_12_13_180500_drop_has_diploma_column.php]: Changed in [39cab07](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8783/commits/39cab07ad29632d9d69c4c806b1dfe03e599558e).

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Sorry, we need to keep some of this for now!

Even if it doesn't correspond directly to a field on the User model, its useful for recruitment team to know whether talent seekers require someone with a diploma.

So the following is all good:
- remove has_diploma from users table
- remove scopes from PoolCandidate and User models
- remove hasDiploma from the user model in the schema
- remove references to hasDiploma from backend user/candidate search functions, since that is meaningless
- remove hasDiploma option from search filters on the admin tables, if its not already gone

However, please restore the following:
- hasDiploma remains part of the ApplicantFilter and PoolCandidateFilter models, in php and in schema
- Talent Seekers can continue to select that they require diplomas (see image below)
- hasDiploma can continue to be attached to the ApplicantFilter submitted with requests
- Request Responders can continue to see whether a request requires diplomas or not (see image below)

Talent Seekers should be able to add education requirement:
image


Education Level visible on admin view of request:
image

**Comment by petertgiles** [comment]
Waiting for #8821 before moving with this one

**Comment by mnigh** [comment]
Closing due to uncertainty.

**Pull request closed without merging**